library/std/src/sys/wasi/os.rs

```diff
@@ -21,6 +21,7 @@ mod libc {
     extern "C" {
         pub fn getcwd(buf: *mut c_char, size: size_t) -> *mut c_char;
         pub fn chdir(dir: *const c_char) -> c_int;
+        pub fn __wasilibc_get_environ() -> *mut *mut c_char;
```

Comment:
So the only cost of this change is that now we need to be sure that this wasilibc function is available via `extern "C"`? And it looks like we already rely on a wasilibc function in library/std/src/sys/wasi/fs.rs so that's nothing new.